yt/python/yt/wrapper/py_wrapper.py

```diff
@@ -756,3 +759,184 @@ def with_formats(func, input_format=None, output_format=None):
+            python_lib_paths
+            if python_lib_paths is not None else
+            [os.path.abspath(path) for path in sys.path]
+        )
+        self._mount = mount
+        self._need_sudo = need_sudo
+
+    def _make_docker_env_args(self, env):
+        return list(itertools.chain(*[
+            ("-e", "{0}={1}".format(key, value))
+            for key, value in env.items()
+        ]))
+
+    def _make_docker_mount_args(self, paths):
+        return list(itertools.chain(*[
+            ("-v", "{0}:{0}".format(path))
+            for path in paths
+        ]))
+
+    def _make_mount_paths(self):
+        # use user provided values
+        if self._mount is not None:
+            return self._mount
+        # if we mount the entire local file system in docker under an arbitrary name
+        # we will break the symlinks,
+        # so we have to find only the paths necessary for work and mount them along the same paths
+        mount_paths = [
+            path
+            for path in [self._cwd, os.path.dirname(self._main_script_path), *self._python_lib_paths]
+            if path is not None and os.path.commonpath([path, self._homedir]) != self._homedir
+        ] + [self._homedir]
+        mount_paths = list(set(mount_paths))
+        mount_paths.sort()
+        return mount_paths
+
+    def _make_docker_platform(self):
+        if self._platform is not None:
+            return "--platform", self._platform
+        return tuple()
+
+    def _make_sudo(self):
+        if self._need_sudo:
+            return ("sudo",)
+        return tuple()
+
+    def make_command(self):
+        env = self._env.copy()
+        # ensure using the same image in all operations
+        env["YT_BASE_LAYER"] = env.get("YT_BASE_LAYER", self._image)
+        # ATTENTION
+        # we use linux-specific separator for PYTHONPATH
+        pythonpath_env = ":".join(self._python_lib_paths)
+
+        mount_paths = self._make_mount_paths()
+        docker_env_args = self._make_docker_env_args(
+            {
+                "CWD": self._cwd,
+                "PYTHONPATH": pythonpath_env,
+                YT_RESPOWNED_IN_CONTAINER_KEY: "1",
+                **env,
+            }
+        )
+        docker_mount_args = self._make_docker_mount_args(mount_paths)
+        command = [
+            *self._make_sudo(),
+            self._docker, "run",
+            *self._make_docker_platform(),
+            "-it", "--rm",
+            *docker_env_args,
+            *docker_mount_args,
+            self._image,
+            "python3", self._main_script_path,
+        ]
+        return command
+
+    def run(self):
+        command = self.make_command()
+        logger.info("Start respawn in docker")
+        logger.info("Command: " + " ".join(command))
+        process = subprocess.Popen(
+            command,
+            stdout=sys.stdout,
+            stderr=sys.stderr,
+            bufsize=1,
+            universal_newlines=True,
```

Comment:
What the idea is behind specifying these options?

```
bufsize=1,
universal_newlines=True,
```

Reply:
This is an old artifact from the time I tried to process stdout/stderr.
I'll just delete it.